rllib/algorithms/algorithm.py

```diff
@@ -196,29 +198,35 @@ def _get_learner_bundles(cf: AlgorithmConfig) -> List[Dict[str, int]]:
             Returns:
                 A list of resource bundles for the learner workers.
             """
-            if cf.num_learners > 0:
-                if cf.num_gpus_per_learner:
-                    learner_bundles = [
-                        {"GPU": cf.num_learners * cf.num_gpus_per_learner}
-                    ]
-                elif cf.num_cpus_per_learner:
-                    learner_bundles = [
-                        {
-                            "CPU": cf.num_learners * cf.num_cpus_per_learner,
-                        }
-                    ]
-            else:
-                learner_bundles = [
-                    {
-                        # sampling and training is not done concurrently when local is
-                        # used, so pick the max.
-                        "CPU": max(
-                            cf.num_cpus_per_learner, cf.num_cpus_for_main_process
+            assert cf.num_learners > 0
+
+            _num = cf.num_learners
+            all_learners = [
+                {
+                    "CPU": _num
+                    * (cf.num_cpus_per_learner if cf.num_gpus_per_learner == 0 else 0),
+                    "GPU": _num
+                    * max(
+                        0,
+                        (
+                            cf.num_gpus_per_learner
+                            - 0.01 * cf.num_aggregator_actors_per_learner
```

Comment:
How does this not fail, if `cf.num_aggregator_actors_per_learner=None`?

Reply:
B/c it's always some int. By default 0.